src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,45 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMUM_AGE = 18;
+    private static final int MINIMUM_LENGTH = 6;
+
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws InvalidUserException {
+        if (user == null) {
+            throw new InvalidUserException("User is null");
+        }
+        if (!isAgeValid(user.getAge())) {
+            throw new InvalidUserException("User age is invalid");
+        }
+        if (!isLoginValid(user.getLogin())) {
+            throw new InvalidUserException("User login is invalid");
+        }
+        if (!isPasswordValid(user.getPassword())) {
+            throw new InvalidUserException("User is null");
+        }
+        if (!isLoginUnique(user.getLogin())) {
+            throw new InvalidUserException("User is null");
```

Comment:
The exception message here is incorrect. It should specify that the login is not unique, not that the user is null. Consider changing the message to something like 'User login is not unique'.